File: src/main/java/com/leijendary/spring/microservicetemplate/config/PropertiesConfiguration.java
package com.leijendary.spring.microservicetemplate.config;

import com.leijendary.spring.microservicetemplate.config.properties.AspectProperties;
import com.leijendary.spring.microservicetemplate.config.properties.AuthProperties;
import com.leijendary.spring.microservicetemplate.config.properties.CorsProperties;
import com.leijendary.spring.microservicetemplate.config.properties.InfoProperties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties({
        AspectProperties.class,
        AuthProperties.class,
        CorsProperties.class,
        InfoProperties.class })
public class PropertiesConfiguration {
}

File: src/main/kotlin/com/leijendary/spring/core/config/PropertiesConfiguration.kt
package com.leijendary.spring.core.config

import com.leijendary.spring.core.config.properties.*
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(
    AuthProperties::class,
    AwsCloudFrontProperties::class,
    AwsS3Properties::class,
    DataSourcePrimaryProperties::class,
    DataSourceReadOnlyProperties::class,
    KafkaTopicProperties::class,
    NumberProperties::class,
    OpenApiProperties::class
)
class PropertiesConfiguration 
